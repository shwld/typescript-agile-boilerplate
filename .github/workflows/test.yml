name: test

on:
  push:
    branches: ["main"]
  pull_request:
    types: [opened, synchronize]

jobs:
  build:
    name: Test
    timeout-minutes: 15
    runs-on: ${{ matrix.os }}
    # To use Remote Caching, uncomment the next lines and follow the steps below.
    # env:
    #  TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
    #  TURBO_TEAM: ${{ secrets.TURBO_TEAM }}
    strategy:
      matrix:
        # os: [ubuntu-latest, macos-latest]
        os: [ubuntu-latest]
    services:
      # Label used to access the service container
      postgres:
        # Docker Hub image
        image: postgres
        # Provide the password for postgres
        env:
          POSTGRES_USER: postgres
          POSTGRES_PASSWORD: password
          POSTGRES_DB: app_test
        # Set health checks to wait until postgres has started
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5

    steps:
      - name: Check out code
        uses: actions/checkout@v2
        with:
          fetch-depth: 2

      - name: Setup Node.js environment
        uses: actions/setup-node@v2
        with:
          node-version: 16
          cache: 'yarn'

      - name: Setup Test
        run: |
          echo 'DATABASE_URL="postgresql://postgres:password@postgres:5432/app_test?schema=public"' > ./infrastructures/db/.env
          echo 'DATABASE_URL="postgresql://postgres:password@postgres:5432/app_test?schema=public"' > ./use-cases/graphql-resolvers/.env
          yarn db:migrate

      - name: Install dependencies
        run: yarn

      - name: Test
        run: yarn test
