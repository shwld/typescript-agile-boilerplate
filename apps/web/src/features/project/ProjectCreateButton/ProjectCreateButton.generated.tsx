import * as Types from '../../../graphql/generated/graphql';

import gql from 'graphql-tag';
import * as Urql from 'urql';
export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
export type ProjectCreateButtonResultFragment = { __typename?: 'Project', id: string, name: string, description: string, privacy: Types.ProjectPrivacy, currentVelocity: number, createdAt: any, accountId: string };

export type ProjectCreateButtonMutationVariables = Types.Exact<{
  input: Types.CreateProjectInput;
}>;


export type ProjectCreateButtonMutation = { __typename?: 'Mutation', createProject: { __typename?: 'CreateProjectSuccessResult', result: { __typename?: 'Project', id: string, name: string, description: string, privacy: Types.ProjectPrivacy, currentVelocity: number, createdAt: any, accountId: string } } | { __typename?: 'InvalidArgumentsResult' } | { __typename?: 'UnauthorizedResult' } };

export const ProjectCreateButtonResultFragmentDoc = gql`
    fragment ProjectCreateButtonResult on Project {
  id
  name
  description
  privacy
  currentVelocity
  createdAt
  accountId
}
    `;
export const ProjectCreateButtonDocument = gql`
    mutation projectCreateButton($input: CreateProjectInput!) {
  createProject(input: $input) {
    ... on CreateProjectSuccessResult {
      result {
        ...ProjectCreateButtonResult
      }
    }
  }
}
    ${ProjectCreateButtonResultFragmentDoc}`;

export function useProjectCreateButtonMutation() {
  return Urql.useMutation<ProjectCreateButtonMutation, ProjectCreateButtonMutationVariables>(ProjectCreateButtonDocument);
};