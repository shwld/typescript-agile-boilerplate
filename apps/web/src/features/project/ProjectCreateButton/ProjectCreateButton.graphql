fragment ProjectCreateButtonResult on Project {
  id
  name
  description
  privacy
  currentVelocity
  createdAt
}

mutation projectCreateButton($input: CreateProjectInput!) {
  createProject(input: $input) {
    ... on CreateProjectSuccessResult {
      result {
        ...ProjectCreateButtonResult
      }
    }
  }
}
