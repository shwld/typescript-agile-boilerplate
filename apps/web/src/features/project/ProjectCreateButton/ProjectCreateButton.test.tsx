import { ProjectCreateButton } from './ProjectCreateButton';
import { render } from '@testing-library/react';
import { MockedUrqlProvider } from '~/test/MockedUrqlProvider';

describe('ProjectCreateButton', () => {
  const renderComponent = () => {
    const renderResult = render(
      <MockedUrqlProvider>
        <ProjectCreateButton accountId="test" />
      </MockedUrqlProvider>
    );
    return renderResult;
  };
  test('Snapshot', () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('success', () => {
    const { getByText } = renderComponent();
    expect(getByText('CREATE PROJECT')).toBeTruthy();
  });
});
