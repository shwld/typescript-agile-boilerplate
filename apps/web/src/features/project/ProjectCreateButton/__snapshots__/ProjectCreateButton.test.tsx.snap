// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCreateButton Snapshot 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-5c2176"
    type="button"
  >
    CREATE PROJECT
  </button>
</DocumentFragment>
`;
